php/blocks/controls/class-image.php

```diff
@@ -57,4 +57,68 @@ public function register_settings() {
 			)
 		);
 	}
+
+	/**
+	 * Validates the value to be made available to the front-end template.
+	 *
+	 * @param mixed $value The value to either make available as a variable or echoed on the front-end template.
+	 * @param bool  $echo Whether this value will be echoed.
+	 * @return mixed $value The value to be made available or echoed on the front-end template.
+	 */
+	public function validate( $value, $echo ) {
+		$is_image = wp_attachment_is( 'image', $value ) && is_numeric( $value );
+
+		if ( $echo ) {
+			return $is_image ? wp_get_attachment_image_url( $value ) : '';
+		} else {
+			return $is_image ? $this->get_attachment_id_from_url( $value ) : false;
+		}
+	}
+
+	/**
+	 * Gets the attachment (image) ID, given its URL.
+	 *
+	 * Based on Micah Wood's solution,
+	 * this creates a \WP_Query to search for the attachment ID.
+	 *
+	 * @see https://wpscholar.com/blog/get-attachment-id-from-wp-image-url/
+	 * @param string $url The URL of the attachment (image).
+	 * @return int|false The attachment ID, or false if none was found.
+	 */
+	public function get_attachment_id_from_url( $url ) {
+		$image_file = basename( $url );
+		$query      = new \WP_Query(
+			array(
+				'post_type'   => 'attachment',
+				'post_status' => 'inherit',
+				'fields'      => 'ids',
+				'meta_query'  => array( // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
```

Comment:
Eventually, we might consider storing the ID of the image in the block, in addition to the URL. Right now, it only stores the URL.

This might help avoid running a meta query to get the ID.

Reply:
@todo: consider using `attachment_url_to_postid()` instead of this helper method.